src/generator/generator.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include "src/algorithms/algorithm.h"
+
+class Generator
+{
+private:
+    std::vector<std::shared_ptr<Algorithm>> algorithms;
+public:
+    const std::vector<std::shared_ptr<Algorithm>>& get_algorithms();
+    void push_back_algorithm(std::shared_ptr<Algorithm>& algorithm);
```

Comment:
push is sufficient

Reply:
I cannot totally agree with that. Maybe "push_back_algorithm" is too descriptive, but "push" doesn't say what that method is doing. Maybe we can meet in the middle and leave "push_algorithm"?

Reply:
If `push` does not say what the method does, then `push_algorithm` does not say it neither. Since algorithm is the type of parameter then you don't have to repeat yourself writing the type of argument in method name. Every modern IDE shows you method prototypes while typing.

Reply:
So basically are you suggesting that I am not using modern IDE?

Reply:
![image](https://user-images.githubusercontent.com/62568699/125707185-8ca2b685-fb90-4f40-b211-cd46cc4b98a8.png)